internal/test/builder/builders.go

```diff
@@ -395,18 +395,19 @@ func (m *MachineDeploymentClassBuilder) Build() *clusterv1.MachineDeploymentClas
 }
 
 // InfrastructureMachineTemplateBuilder holds the variables and objects needed to build an InfrastructureMachineTemplate.
-// +kubebuilder:object:generate=false
 type InfrastructureMachineTemplateBuilder struct {
-	namespace  string
-	name       string
-	specFields map[string]interface{}
+	obj *unstructured.Unstructured
 }
 
 // InfrastructureMachineTemplate creates an InfrastructureMachineTemplateBuilder with the given name and namespace.
 func InfrastructureMachineTemplate(namespace, name string) *InfrastructureMachineTemplateBuilder {
+	obj := &unstructured.Unstructured{}
```

Comment:
Next we have to change all the functions that act on the builder so that they no longer fill variables, but instead add the fields directly to the obj.

In the first function (which is used to initialize the builder) we can set Name, Namespace, APIVersion and Kind using the accessor methods for unstructured.

We can then return the builder with the object set inside in the .obj field.